libs/cms/src/lib/models/utils.ts

```diff
@@ -6,3 +15,61 @@ export const getRelativeUrl = (url: string) => {
   }
   return urlObject.href
 }
+
+type LinkType = Omit<ILink | ILinkedPage, 'update' | 'toPlainObject'>
+
+export const mapReferenceLink = (link?: LinkType): Link | null => {
+  if (!link || !link.sys?.contentType?.sys?.id) return null
+
+  const contentTypeId = link.sys.contentType.sys.id
+
+  if (contentTypeId === 'linkedPage') {
+    return generateOrganizationSubpageLinkFromLinkedPage(link as ILinkedPage)
+  }
+
+  if (contentTypeId === 'link') {
+    return mapLink(link as ILink)
+  }
+
+  return null
+}
+
+const generateOrganizationSubpageLinkFromLinkedPage = (
+  linkedPage: ILinkedPage,
+): Link | null => {
+  const contentTypeId = linkedPage.fields.page.sys.contentType.sys.id
+  let slug: string
```

Comment:
Let's initialize this variable here as well just in case someone comes in the future and changes this function